src/csharp/Grpc.Auth/GoogleAuthInterceptors.cs

```diff
@@ -43,29 +45,22 @@ public static class GoogleAuthInterceptors
         /// <returns>The interceptor.</returns>
         public static AsyncAuthInterceptor FromCredential(ITokenAccess credential)
         {
-            if(credential is ITokenAccessWithExtraInformation credentialWithExtra)
-            {
-                return FromCredential(credentialWithExtra);
-            }
-            
-            return new AsyncAuthInterceptor(async (context, metadata) =>
+            AsyncAuthInterceptor interceptors = new AsyncAuthInterceptor(async (context, metadata) => 
             {
-                await AddBearerTokenHeader(credential, context, metadata).ConfigureAwait(false);
+                var accessToken = await credential.GetAccessTokenForRequestAsync(context.ServiceUrl, CancellationToken.None).ConfigureAwait(false);
+                metadata.Add(CreateBearerTokenHeader(accessToken));
             });
-        }
 
-        /// <summary>
-        /// Creates an <see cref="AsyncAuthInterceptor"/> that will obtain access token and associated information
-        /// from any credential type that implements <c>ITokenAccessWithExtraInformation</c>.
-        /// </summary>
-        /// <param name="credential">The credential to use to obtain access tokens.</param>
-        /// <returns>The interceptor.</returns>
-        public static AsyncAuthInterceptor FromCredential(ITokenAccessWithExtraInformation credential)
-        {
-            return new AsyncAuthInterceptor(async (context, metadata) => {
-                await AddBearerTokenHeader(credential, context, metadata).ConfigureAwait(false);
-                AddExtraInforHeaders(credential, context, metadata);
-            });
+            if (credential is IProjectImpersonatorTokenAccess impersonator)
```

Comment:
Somewhat dirty implementation that keeps the surface cleaner. If we need to add more similar info in the future, we can add more interfaces in Google.Apis.Auth and the corresponding ifs here.